erigon-lib/diagnostics/utils.go

```diff
@@ -44,8 +45,21 @@ func InitStagesFromList(list []string) []SyncStage {
 			ID:        stage,
 			State:     Queued,
 			SubStages: []SyncSubStage{},
+			Stats:     SyncStageStats{},
 		})
 	}
 
 	return stages
 }
+
+func CalculateTime(amountLeft, rate uint64) string {
+	if rate == 0 {
+		return "999hrs:99m"
+	}
+	timeLeftInSeconds := amountLeft / rate
+
+	hours := timeLeftInSeconds / 3600
```

Comment:
time.Duration(timeLeftInSeconds) type has method String to pretty print durations btw